### PR TITLE
Fix comments query

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR fixes the comment query to be done in the DB.

### CODE REVIEW

File: skyportal/handlers/api/source.py [hunk 1]
```diff
@@ -733,6 +733,38 @@ def get_sources(
             raise ValueError(
                 "Invalid commentsFilter value -- must provide at least one string value"
             )
+
+        comment_query = Comment.select(session.user_or_token).where(
+            Comment.text.in_(comments_filter)
+        )
```
Comment (guynir42) [inline skyportal/handlers/api/source.py:739]: Why not filter on all the comment related stuff in one subquery?

File: skyportal/handlers/api/source.py [hunk 2]
```diff
@@ -1004,49 +1054,6 @@ def get_sources(
                                 break
                 if not passes_filter:
                     continue
-            if (
```
Comment (guynir42) [inline skyportal/handlers/api/source.py:1007]: I'm happy to see this go into the direct query of the DB. 

We should think about making a partial text search of the comments at some point. I'm sure it is something that's pretty commonly done in similar situations.

**Review by guynir42 — COMMENTED** [review]
Why have multiple subqueries?

File: skyportal/handlers/api/source.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py:1007]: Yeah definitely, just not obvious how to do easily.

File: skyportal/handlers/api/source.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/source.py]: Should be fixed!

**Pull request merged**